From 5a7994b56ae2a5fb79562be1c7362894f5454a64 Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Wed, 13 Jun 2012 16:30:41 +0900
Subject: [PATCH 54/73] mediabus: add new packings for 16-bit samples

16 bits transferred in 1 16-bit samples cannot be described by any
of the existing SOC_MBUS_PACKING_* macros.
This patch adds new SOC_MBUS_PACKING_1X16_PADHI and
SOC_MBUS_PACKING_1X16_PADLO to describe such formats and extends
soc_mbus_samples_per_pixel() to support it.
---
 drivers/media/video/soc_mediabus.c |    2 ++
 include/media/soc_mediabus.h       |    5 +++++
 2 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/soc_mediabus.c b/drivers/media/video/soc_mediabus.c
index 866ab8d..086730b 100644
--- a/drivers/media/video/soc_mediabus.c
+++ b/drivers/media/video/soc_mediabus.c
@@ -160,6 +160,8 @@ s32 soc_mbus_bytes_per_line(u32 width, const struct soc_mbus_pixelfmt *mf)
 		return width * mf->bits_per_sample / 8;
 	case SOC_MBUS_PACKING_2X8_PADHI:
 	case SOC_MBUS_PACKING_2X8_PADLO:
+	case SOC_MBUS_PACKING_1X16_PADHI:
+	case SOC_MBUS_PACKING_1X16_PADLO:
 	case SOC_MBUS_PACKING_EXTEND16:
 		return width * 2;
 	case SOC_MBUS_PACKING_EXTEND32:
diff --git a/include/media/soc_mediabus.h b/include/media/soc_mediabus.h
index 25e7b6d..c86efa6 100644
--- a/include/media/soc_mediabus.h
+++ b/include/media/soc_mediabus.h
@@ -21,6 +21,9 @@
  * @SOC_MBUS_PACKING_2X8_PADHI:	16 bits transferred in 2 8-bit samples, in the
  *				possibly incomplete byte high bits are padding
  * @SOC_MBUS_PACKING_2X8_PADLO:	as above, but low bits are padding
+ * @SOC_MBUS_PACKING_1X16_PADHI: 16 bits transferred in 1 16-bit samples, in the
+ *				possibly incomplete byte high bits are padding
+ * @SOC_MBUS_PACKING_1X16_PADLO: as above, but low bits are padding
  * @SOC_MBUS_PACKING_EXTEND16:	sample width (e.g., 10 bits) has to be extended
  *				to 16 bits
  * @SOC_MBUS_PACKING_EXTEND32:  sample width (e.g., 24 bits) has to be extended
@@ -30,6 +33,8 @@ enum soc_mbus_packing {
 	SOC_MBUS_PACKING_NONE,
 	SOC_MBUS_PACKING_2X8_PADHI,
 	SOC_MBUS_PACKING_2X8_PADLO,
+	SOC_MBUS_PACKING_1X16_PADHI,
+	SOC_MBUS_PACKING_1X16_PADLO,
 	SOC_MBUS_PACKING_EXTEND16,
 	SOC_MBUS_PACKING_EXTEND32,
 };
-- 
1.7.0.4

